Feature: Autogenerate Telegraf docs from influxdata/telegraf repo.

jstirnaman opened the pull request:
Adds templates from the `telegraf-internal` repo to permanently reside in `/docs-v2/telegraf-build`.

Supports auto-generating Telegraf documentation from the Telegraf release process.
The Telegraf release workflow will trigger the build of a `docs` binary. When run, the script extracts README file content from the Telegraf release, applies the templates in this PR, and generates a pull request to the `docs-v2` repo.

Depends on PR https://github.com/influxdata/telegraf-internal/pull/488

<img width="1237" alt="image" src="https://github.com/user-attachments/assets/7c1d8da0-a335-4ca2-89a6-7d776b834dc3" />
<img width="1483" alt="image" src="https://github.com/user-attachments/assets/e68aabd7-af04-4198-bb5b-d341fe241da9" />

sanderson commented on build-telegraf/plugins.sh:
```diff
@@ -0,0 +1,112 @@
+
+EOF
+  else
+    echo "Plugin directory not found: $plugin_dir"
```
We currently have 20 external plugins that are listed on the plugins page but are not part of the Telegraf repo. How do we want to handle these? @srebhan, do you have any thoughts?

sanderson commented on layouts/shortcodes/telegraf/plugins.html:
```diff
@@ -23,5 +23,6 @@ <h3 id="{{ $type }}-{{ .id }}">{{ .name }}</h3>
     <a class="btn github-link" href="{{ if .link }}{{ .link }}{{ else }}https://github.com/influxdata/telegraf/blob/release-{{ $latestTelegrafVersion }}/plugins/{{ $type }}s/{{ .id }}/README.md{{ end }}" target="_blank">
       <span class="icon-github"></span> <span class="hide">View</span>
     </a>
+    <div>{{ .Site.GetPage "/telegraf/v1/input-plugins/apache/_index.md" }}</div>
```
I assume you'll iterate over the page data to do something here?

sanderson commented on build-telegraf/plugins.sh:
```diff
@@ -0,0 +1,112 @@
+  telegraf_v1_ref:
+    parent: $parent
+    name: $(basename "$plugin_dir")
+tags: [$(basename "$plugin_dir")]
```
One thing that would be nice to account for here is `introduced` and `deprecated` versions. Were you thinking we'd do that by injecting frontmatter or adding that frontmatter to the readmes?

jstirnaman committed:
WIP: Test generating Telegraf docs in a local environment (Dockerfile, README)

jstirnaman committed:
feat(telegraf): Use metadata from autogenerated Telegraf docs in template frontmatter.
- Use plugin type and version to build related links
- Insert tag list into frontmatter tags
- Rename metadata in content body

jstirnaman committed:
Merge branch 'master' into automate-telegraf-docs

jstirnaman commented:
@srebhan Merging to support PR influxdata/telegraf-internal#488
